app/services/course/student_statistics_service.rb

```diff
@@ -18,11 +18,15 @@ def initialize(tutors)
   # @return [Array<CourseUser>]
   def tutors_of(course_user)
     course_user.groups.map do |group|
-      @tutors.select do |tutor|
-        tutor.group_users.select do |group_user|
-          group_user.manager? && group_user.group_id == group.id
-        end.present?
-      end
-    end.flatten.uniq
+      group_managers_hash[group]
+    end.flatten.map(&:course_user).uniq
+  end
+
+  # Maps groups to their managers
+  #
+  # @return [Hash{Course::Group => Array<Course::GroupUser>}]
+  def group_managers_hash
+    @group_managers_hash ||=
+      @tutors.map(&:group_users).flatten.select(&:manager?).group_by(&:group)
```

Comment:
Not sure whether it matters if we use `group_by(&:group)` vs `group_by(&:group_id)`?